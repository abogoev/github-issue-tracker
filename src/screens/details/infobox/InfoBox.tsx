import React, { VFC } from "react";
import { StyleSheet, Text, View, Image, ViewProps } from "react-native";
import theme from "../../../theme/theme";
import { Issue, WithoutChildren } from "../../../types";

interface Props
  extends WithoutChildren<ViewProps>,
    Pick<Issue, "body" | "user"> {
  createdAt: Date;
}

const Comment: VFC<Props> = ({ body, user, style, createdAt, ...props }) => {
  return (
    <View style={[styles.container, style]} {...props}>
      <Image
        accessibilityIgnoresInvertColors
        source={{ uri: user.avatar_url }}
        style={styles.image}
      />
      <View style={styles.messageContainer}>
        <View style={styles.topMessageContainer}>
          <Text>{user.login}</Text>
          <Text>commented on {createdAt.toLocaleString()}</Text>
        </View>
        <View style={styles.bottomMessageContainer}>
          <Text>{body}</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
  },
  image: {
    width: 40,
    height: 40,
    borderRadius: 20,
    marginRight: theme.spacing.medium,
  },
  messageContainer: {
    flex: 1,
    borderColor: theme.palette.neutral,
    borderWidth: 1,
    borderRadius: theme.borderRadius,
  },
  topMessageContainer: {
    minHeight: 39,
    backgroundColor: theme.palette.lightGrey,
    paddingHorizontal: theme.spacing.medium,
    paddingVertical: theme.spacing.small,
    borderTopLeftRadius: theme.borderRadius,
    borderTopRightRadius: theme.borderRadius,
    borderBottomWidth: 1,
    borderColor: theme.palette.neutral,
  },
  bottomMessageContainer: {
    minHeight: 53,
    padding: theme.spacing.medium,
  },
});

export default Comment;
