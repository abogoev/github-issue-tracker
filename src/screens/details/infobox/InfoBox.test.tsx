import React from "react";
import InfoBox from "./InfoBox";
import { create } from "react-test-renderer";

describe("InfoBox", () => {
  it("should render correctly", () => {
    const component = create(
      <InfoBox
        body="Body"
        user={{ avatar_url: "", login: "username" }}
        createdAt={new Date("2000-01-01")}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
