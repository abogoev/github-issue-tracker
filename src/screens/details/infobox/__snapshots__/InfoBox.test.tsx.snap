// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityIgnoresInvertColors={true}
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginRight": 16,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(246, 248, 250)",
          "borderBottomWidth": 1,
          "borderColor": "rgb(208, 215, 222)",
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
          "minHeight": 39,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        username
      </Text>
      <Text>
        commented on 
        01/01/2000, 02:00:00
      </Text>
    </View>
    <View
      style={
        Object {
          "minHeight": 53,
          "padding": 16,
        }
      }
    >
      <Text>
        Body
      </Text>
    </View>
  </View>
</View>
`;
